### ISSUE TITLE
Additional fields?

### ISSUE THREAD

**Issue by tkerns** [alperenersoy/filament-export#14]
This appears to only export columns that are visible in the table (ie hidden columns, or columns toggled off are not available).
Is there a way to  export all fields?  Or an array of fields, or all except?
Perhaps I just missed something in the docs.
Great plugin though!

**Comment by alperenersoy** [comment]
thanks! it is not possible right now. we use filament's getColumns method and it filters toggled ones. somebody else also asked for this. I will try to figure a way out.

**Comment by tkerns** [comment]
Yes, I saw that when I went source diving.  And I noticed filament's getColumns method does not include hidden columns (I was hoping maybe you could pass an argument into getColumns to show all, but no joy).
As a start, maybe add the ability to pass an array of columns, that if it exists, use that instead of getColumns. 
I didn't want to dig too deep until I heard from you if maybe there was something I missed.  Thanks for the prompt response!

**Comment by alperenersoy** [comment]
I guess we can also use `getCachedTableColumns` directly but filament is evolving rapidly and function names and other things change sometimes. so `getColumns` seems safer. and yes we can add sth like `->only([])` and `->except([])` to filter and also maybe sth like `->with([])` to solve hidden and unlisted columns. it would be nice to have these options. I am working on some performance improvements for this plugin and I will take care of this too asap.

**Comment by tkerns** [comment]
Yes, I replaced the `getColumns` with `getCachedTableColumns` in 2 places in `FilamentExport` and it seemed to pull in all fields/labels from the form.  So that might be a good start.  But I understand about filament evolving rapidly, and using undocumented methods might be risky :-)

**Comment by alperenersoy** [comment]
Yes, you are right. I've been working on this and yes `getCachedTableColumns` gets toggled ones but filters `->hidden()` columns. I am not sure what to do tbh.

**Comment by tkerns** [comment]
Well, what I did, is change the `getColumns` with `getCachedTableColumns` in 2 places in `FilamentExport` and then in my filament resource, in the table, I put all the columns that I wanted in my export in the table.  The ones that I did not want to appear in my table as toggleable, and then made them toggled off by default 
`Tables\Columns\TextColumn::make('comments')->toggleable(isToggledHiddenByDefault: true),`
That seems to work well for me.

**Comment by alperenersoy** [comment]
sorry for the late response. I will add something for this and inform you about that. But I think we should somehow be able to add ->hidden() columns too.

**Comment by tkerns** [comment]
Is there a reason you don't want to use `getCachedTableColumns` instead of `getColumns` to allow toggleable fields to be included in the export?  It would sure be nice to use this plugin, instead of having to maintain a forked repo :-).  I completely understand if you don't want to go that direction, I was just curious if you had a reason.

**Comment by alperenersoy** [comment]
oh sorry. I've been a little bit busy lately and wanted to keep it as a future plan but if it is essential for you I will add it as an option asap.

**Comment by tkerns** [comment]
Thank you!  I really like this plugin as compared to other options, and so I have just been maintaining a version with the changes I outlined above.  It has been working well for me.  I totally understand about being busy, and have great respect for folks who put effort into open source software.  You don't need to do it immediately just for my benefit.  I just wanted to make sure it was still on the table.  I would be happy to make a pull request with the changes I made, but in all reality, it is only changing  `getColumns` with `getCachedTableColumns` in 2 places in FilamentExport, and it was your suggestion that I followed.

**Comment by alperenersoy** [comment]
hey!

added ->withHiddenColumns() function to actions and released it. I hope that works for you. actually showing them by default and unchecked in the filtering columns checkbox list is a better idea but let me think about it.

**Comment by tkerns** [comment]
Hey, thanks so much!  I personally like them checked by default in the filtering columns checkbox list, as the only reasons I am even adding those toggleable fields to the table is so they can be included in the export.
I gave it a quick test, and it worked perfectly.
I looked through your implementation, and you did it much nicer that I would have. :-) I like the idea of adding it as a concern, and then only using `getCachedTableColumns`  if specifically asked to use `withHiddenColumns`.
